lib/src/values/logic_value.dart

```diff
@@ -678,6 +678,45 @@ abstract class LogicValue {
   // ignore: avoid_dynamic_calls
   LogicValue operator %(dynamic other) => _doMath(other, (a, b) => a % b);
 
+  /// Ceil of log base 2 operation.
+  ///
+  /// Returns ceil of log base 2 of valid input.
+  /// Returns `x` if any bit of input is invalid.
+  LogicValue clog2() => _doUnaryMath(_clog2);
+
+  /// Executes mathematical operations on single [LogicValue]
+  ///
+  /// Handles invalid input and do proper type conversion to required types
+  LogicValue _doUnaryMath(dynamic Function(dynamic num, int width) op) {
```

Comment:
`num` is a built-in type in Dart, so it's a little confusing to have variables throughout named `num`.  Can we try a different variable name?